Sources/Wrappers/AngouriMath.FSharp/Matrices.fs

```diff
@@ -3,45 +3,101 @@
 open AngouriMath
 open AngouriMath.FSharp.Core
 open AngouriMath.FSharp.Functions
+open System.Linq
 
-let (|*) a b = (parsed a * parsed b).InnerSimplified |> asMatrix
+/// Represents the length type of a matrix
+type Lengths =
+    | Any
+    | Invalid
+    | Fixed of int
 
-let (|**) a b = (parsed a ** parsed b).InnerSimplified |> asMatrix
+/// Creates a matrix from a list of lists
+/// of objects (which are parsed into Entities)
+let matrix x = 
+    let rec columnCount (x : 'T list list) =
+        match x with
+        | [] -> Any
+        | hd::tl ->
+            match columnCount tl with
+            | Any -> Fixed(hd.Length)
+            | Invalid -> Invalid
+            | Fixed len -> if len = hd.Length then Fixed(len) else Invalid
+            
+    let parseListOfLists li =
+        [ for row in li do yield [ for el in row do yield (parsed el) ] ]
 
-let (|+) a b = (parsed a + parsed b).InnerSimplified |> asMatrix
+    match columnCount x with
+    | Any | Invalid -> raise ParseException
+    | Fixed _ -> MathS.Matrix(array2D (parseListOfLists x))
 
-let (|-) a b = (parsed a - parsed b).InnerSimplified |> asMatrix
-
-let (|/) a b = (parsed a / parsed b).InnerSimplified |> asMatrix
+/// Creates a column vector from a 1-dimensional list
+let vector li =
+    MathS.Vector([ for el in li do yield parsed el ].ToArray())
```

Comment:
```suggestion
    MathS.Vector [| for el in li do parsed el |]
```

Reply:
Fair. These are mostly related to how I wrote it, so it's not exactly the problem of @ryyanmapes , but it would be nice to have them fixed in this PR.